### ISSUE TITLE
Implement app containerization

### ISSUE THREAD

**Issue by matthewwangg** [sinkovit/Spatial-ecology#160]
I've been working on containerizing the app and have created a Dockerfile for the Shiny app.

**Comment by monaw** [comment]
@matthewwangg, is it possible for the app to write the user's stadiamap key to a file in the container and then read it back next time the container is run?

**Comment by matthewwangg** [comment]
No, because the file wouldn't be kept when the container is stopped, but a .env file could be kept on the user's local directory and the container could be ran with the tag --env-file .env to allow the key to be loaded in. The container branch supports this as currently constructed

**Comment by monaw** [comment]
thanks @matthewwangg. Where is this "user's local directory"?  do we need to configure the image to allow for this setup?  also which version of R is in the container?

**Comment by matthewwangg** [comment]
The user's local directory is wherever they run the container from on the local machine. No need to configure the image, I already modified the app accordingly, and yes the variable would be available in the app. The current version is 4.4.1 but I will be modifying that

**Comment by monaw** [comment]
can you describe the modification you did a bit more.  how does the user setup their Stadia map API key?  what are you going to change the R version to?

**Comment by matthewwangg** [comment]
The modification is to util.R, it pulls an environment variable instead of a file now, I'm testing with R versions still, plus need to be prepared for MKDE 0.4

**Comment by monaw** [comment]
what's the docker command line you use to run your container?  i'll add it to the documentation...

**Comment by sinkovit** [comment]
Hi Mona, I'll let Matt answer that. I've been trying to run through Docker desktop and getting stuck

**Comment by matthewwangg** [comment]
@monaw It would look something like `docker run --env-file .env -p 3838:3838 <name>` with the .env file being present in the directory locally that you run the container from

**Comment by matthewwangg** [comment]
@sinkovit What issues are appearing for Docker Desktop? I have been running it through CLI, but I could look into that

**Comment by monaw** [comment]
@sinkovit , i got it to work with Docker Desktop on my mac.  i created a file named env and put "STADIA_MAPS_API_KEY=<api_key_value>" in the file. then i started the container in the terminal:

`docker run --env-file env -p 3838:3838 ghcr.io/matthewwangg/spatial-ecology-gateway-dev`

when you see:

```
[...]
Listening on http://0.0.0.0:3838
ℹ © Stadia Maps © Stamen Design © OpenMapTiles © OpenStreetMap contributors.

```
then you can open your browser to http://localhost:3838 and you should see the app running.

alternatively in your Docker Desktop, click on the container port (see attached image) and it should open your browser to it 

![Screenshot 2024-12-10 at 3 37 49 PM](https://github.com/user-attachments/assets/c3f92ab2-33c6-44b1-acf0-1de7a9f4a20b)

i will add these instructions to our README...

**Comment by matthewwangg** [comment]
Note that the updated package is now called spatial-ecology-gateway-secure
